src/traces/histogram/calc.js

```diff
@@ -232,8 +232,8 @@ function calcAllAutoBins(gd, trace, pa, mainData, _overlayEdgeCase) {
 
     // all but the first trace in this group has already been marked finished
     // clear this flag, so next time we run calc we will run autobin again
-    if(trace._autoBinFinished) {
-        delete trace._autoBinFinished;
+    if(trace['_' + mainData + 'autoBinFinished']) {
```

Comment:
Fixup from https://github.com/plotly/plotly.js/issues/3909

As now both `histogram` and `histogram2d*` traces go through `calcAllAutoBins`, we need to split the "has autoBin finished` logic into x and y values.